src/creatures/players/player.cpp

```diff
@@ -2469,6 +2469,91 @@ void Player::death(Creature* lastHitCreature)
 		onIdleStatus();
 		sendStats();
 	}
+	despawn();
+}
+
+bool Player::spawn()
+{
+	setDead(false);
+
+	const Position& pos = getLoginPosition();
+
+	if (!g_game.map.placeCreature(pos, this, false, true)) {
+		return false;
+	}
+
+	SpectatorHashSet spectators;
+	g_game.map.getSpectators(spectators, position, false, true);
+	for (Creature* spectator : spectators) {
+		if (Player* tmpPlayer = spectator->getPlayer()) {
+			tmpPlayer->sendCreatureAppear(this, pos, true);
+		}
+	}
+
+	for (Creature* spectator : spectators) {
+	spectator->onCreatureAppear(this, true);
+	}
+
+	getParent()->postAddNotification(this, nullptr, 0);
+	g_game.addCreatureCheck(this);
+
+	addList();
+	return true;
+}
+
+void Player::despawn()
+{
+	if (isDead()) {
+		return;
+	}
+
+	listWalkDir.clear();
+	stopEventWalk();
+	onWalkAborted();
+
+	// remove check
+	g_game.removeCreatureCheck(this);
+
+	// remove from map
+	Tile* tile = getTile();
```

Comment:
Here need have a check pointer: 
```suggestion
	Tile* tile = getTile();
	if (!tile) {
	    return;
	}
```